src/observable-hook.ts

```diff
@@ -0,0 +1,37 @@
+import { TRAM_OBSERVABLE_STORE, TRAM_HOOK_KEY } from './engine-names';
+import { getObservableStore } from './observable-store';
+import { getWorkingKeyValue, incrementWorkingKeyBranch } from './working-key';
+
+import { StoreObject } from './types';
+
+/**
+ * Shared source code for both observable hooks, useStore, and useGlobalStore.
+ * This hook exposes a globally stored value (in either case), that can cause the component
+ * to update when a subfield of that value is updated.
+ *
+ * It has a similar interface to React's useState
+ */
+export default <Store extends StoreObject>(key?: string, value?: Store): Store => {
```

Comment:
Previously this function verified the type of the string and value, now that can be verified with Typescript! JS users will still get annotations in Typescript capable editors.